cpp/ql/lib/semmle/code/cpp/models/implementations/SmartPointer.qll

```diff
@@ -168,3 +168,58 @@ private class SmartPtrSetterFunction extends MemberFunction, AliasFunction, Side
     )
   }
 }
+
+/** A destructor assocaited with a smart pointer. */
+private class SmartPtrDestructor extends Destructor, SideEffectFunction, AliasFunction {
+  SmartPtr declaringType;
+
+  SmartPtrDestructor() {
+    declaringType = this.getDeclaringType() and not this.isFromUninstantiatedTemplate(_)
+  }
+
+  /**
+   * Gets the destructor associated with the base type of this smart pointer.
+   */
+  private Destructor getBaseTypeDestructor() {
+    // TODO: Check if this is join ordered correctly.
+    result.getDeclaringType() = declaringType.getBaseType()
```

Comment:
Did you check this? Because I was kinda hoping that you would 😂 I think we should just start a fresh DCA run on this PR, and if nothing stands out here we should just delete the `TODO` comment.